syncing/src/test/kotlin/maru/syncing/beaconchain/CLSyncServiceImplTest.kt

```diff
@@ -0,0 +1,493 @@
+
+    // Fail the first two calls to getPeers() to simulate failure getting peers and not downloading blocks
+    var retries = 0
+    doAnswer {
+      if (retries < 2) {
+        retries++
+        throw IllegalStateException("Simulated failure for testing")
+      } else {
+        it.callRealMethod()
+      }
+    }.whenever(peerLookup).getPeers()
+
+    val metricsFacade = mock(MetricsFacade::class.java)
+    val retriesCounter = mock(Counter::class.java)
+    whenever(metricsFacade.createCounter(any(), any(), any(), any())).thenReturn(retriesCounter)
+
+    val synced = AtomicBoolean(false)
+    clSyncService.setSyncTarget(100uL)
+    clSyncService.onSyncComplete { synced.set(true) }
+    awaitUntilAsserted { assertThat(synced).isTrue() }
+
+    assertThat(beaconChain1.getLatestBeaconState().latestBeaconBlockHeader.number).isEqualTo(100uL)
+    assertThat(beaconChain1.getLatestBeaconState()).isEqualTo(beaconChain2.getLatestBeaconState())
+    for (i in 0uL..100uL) {
+      assertThat(beaconChain1.getSealedBeaconBlock(i)).isEqualTo(beaconChain2.getSealedBeaconBlock(i))
+      assertThat(beaconChain1.getBeaconState(i)).isEqualTo(beaconChain2.getBeaconState(i))
+    }
+
+    verify(retriesCounter, times(retries)).increment()
+  }
+
+  @Test
+  fun `sync target set to same target returns immediately`() {
+    // sync to block 50
+    val synced = AtomicBoolean(false)
+    clSyncService.setSyncTarget(50uL)
+    clSyncService.onSyncComplete { synced.set(true) }
+    awaitUntilAsserted { assertThat(synced).isTrue() }
+
+    assertThat(beaconChain1.getLatestBeaconState().latestBeaconBlockHeader.number).isEqualTo(50uL)
+    assertThat(beaconChain1.getLatestBeaconState()).isEqualTo(beaconChain2.getBeaconState(50uL))
+    for (i in 1uL..50uL) {
+      assertThat(beaconChain1.getSealedBeaconBlock(i)).isEqualTo(beaconChain2.getSealedBeaconBlock(i))
+      assertThat(beaconChain1.getBeaconState(i)).isEqualTo(beaconChain2.getBeaconState(i))
+    }
+
+    // update the sync target to 50 again
+    synced.set(false)
+    reset(beaconChain1, beaconChain2)
+    clSyncService.setSyncTarget(50uL)
+    awaitUntilAsserted { assertThat(synced).isTrue() }
+
+    // Ensure the state remains unchanged
+    assertThat(beaconChain1.getLatestBeaconState().latestBeaconBlockHeader.number).isEqualTo(50uL)
+    assertThat(beaconChain1.getLatestBeaconState()).isEqualTo(beaconChain2.getBeaconState(50uL))
+    for (i in 1uL..50uL) {
+      assertThat(beaconChain1.getSealedBeaconBlock(i)).isEqualTo(beaconChain2.getSealedBeaconBlock(i))
+      assertThat(beaconChain1.getBeaconState(i)).isEqualTo(beaconChain2.getBeaconState(i))
+    }
+
+    // Verify that beaconChain for node1 was not updated, and there are no reads on node2 beaconChain
+    verify(beaconChain1, never()).newUpdater()
+    verify(beaconChain2, never()).getSealedBeaconBlocks(any(), any())
+  }
+
+  @Test
+  fun `sync target set to older target returns immediately`() {
```

Comment:
Can we please add the following test, similar to this one:
1. The source beacon chain only has blocks until 100
2. The target is set to 150
3. The target is set to 100
The expectation that the target beacon chain is synced up to a block 100

?